@xen-orchestra/xapi/vdi.mjs

```diff
@@ -58,6 +58,28 @@ class Vdi {
     )
   }
 
+  async destroyCloudInitConfig(vdiRef, { timeLimit = Date.now() + 10 * 60 * 1000 } = {}) {
+    const vbdRef = (await this.getField('VDI', vdiRef, 'VBDs'))[0]
+    const vmRef = await this.getField('VBD', vbdRef, 'VM')
+
+    await this.waitObjectState(vmRef, vm => vm.power_state === 'Running', {
+      timeout: timeLimit - Date.now(),
+    })
+
+    const vm = await this.getRecord('VM', vmRef)
+    await this.waitObjectState(vm.guest_metrics, gm => gm?.PV_drivers_version?.major !== undefined, {
```

Comment:
It has recently be suggested that we should avoid using `PV_drivers_version` to check if the VM is booted, but I don't remember how these discussions ended. Maybe @fbeauchamp knows?

Reply:
I think this is not a problem here since we can't create a hvm VM from the create VM form